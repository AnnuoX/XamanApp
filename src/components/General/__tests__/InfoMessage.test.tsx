/**
 * @format
 */

import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import { InfoMessage } from '..';

describe('[InfoMessage]', () => {
    describe('renders correctly', () => {
        it('Info', () => {
            const tree = renderer.create(<InfoMessage label="label" type="info" />).toJSON();
            expect(tree).toMatchSnapshot();
        });
        it('Warning', () => {
            const tree = renderer.create(<InfoMessage label="label" type="warning" />).toJSON();
            expect(tree).toMatchSnapshot();
        });
        it('Error', () => {
            const tree = renderer.create(<InfoMessage label="label" type="error" />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('should render with icon', () => {
        it('Info', () => {
            const tree = renderer.create(<InfoMessage label="label" icon="IconInfo" type="info" />).toJSON();
            expect(tree).toMatchSnapshot();
        });
        it('Warning', () => {
            const tree = renderer.create(<InfoMessage label="label" icon="IconInfo" type="warning" />).toJSON();
            expect(tree).toMatchSnapshot();
        });
        it('Error', () => {
            const tree = renderer.create(<InfoMessage label="label" icon="IconInfo" type="error" />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
