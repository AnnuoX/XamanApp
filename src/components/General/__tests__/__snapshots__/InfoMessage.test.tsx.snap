// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[InfoMessage] renders correctly Error 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 15,
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
      undefined,
      null,
      null,
      Object {
        "backgroundColor": "#FFF9F9",
        "borderColor": "#FFF9F9",
        "tintColor": "#FF5B5B",
      },
      null,
      null,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "paddingVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Bold",
            "fontSize": 30,
            "textAlign": "center",
          },
          null,
          null,
          Object {
            "color": "#FF5B5B",
          },
          null,
          null,
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      label
    </Text>
  </View>
</View>
`;

exports[`[InfoMessage] renders correctly Info 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 15,
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
      undefined,
      Object {
        "backgroundColor": "#F0F3FF",
        "borderColor": "#3052FF",
        "tintColor": "#3052FF",
      },
      null,
      null,
      null,
      null,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "paddingVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Bold",
            "fontSize": 30,
            "textAlign": "center",
          },
          Object {
            "color": "#3052FF",
          },
          null,
          null,
          null,
          null,
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      label
    </Text>
  </View>
</View>
`;

exports[`[InfoMessage] renders correctly Warning 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 15,
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
      undefined,
      null,
      Object {
        "backgroundColor": "#FFF9ED",
        "borderColor": "#F8BF4C",
        "tintColor": "#F8BF4C",
      },
      null,
      null,
      null,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "paddingVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Bold",
            "fontSize": 30,
            "textAlign": "center",
          },
          null,
          Object {
            "color": "#F8BF4C",
          },
          null,
          null,
          null,
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      label
    </Text>
  </View>
</View>
`;

exports[`[InfoMessage] should render with icon Error 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 15,
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
      undefined,
      null,
      null,
      Object {
        "backgroundColor": "#FFF9F9",
        "borderColor": "#FFF9F9",
        "tintColor": "#FF5B5B",
      },
      null,
      null,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "paddingRight": 10,
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "IconInfo",
        }
      }
      style={
        Array [
          Object {
            "height": 31.428571428571427,
            "width": 31.428571428571427,
          },
          Array [
            null,
            null,
            Object {
              "tintColor": "#FF5B5B",
            },
            null,
            null,
          ],
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "paddingVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Bold",
            "fontSize": 30,
            "textAlign": "center",
          },
          null,
          null,
          Object {
            "color": "#FF5B5B",
          },
          null,
          null,
          Object {
            "textAlign": "left",
          },
        ]
      }
    >
      label
    </Text>
  </View>
</View>
`;

exports[`[InfoMessage] should render with icon Info 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 15,
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
      undefined,
      Object {
        "backgroundColor": "#F0F3FF",
        "borderColor": "#3052FF",
        "tintColor": "#3052FF",
      },
      null,
      null,
      null,
      null,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "paddingRight": 10,
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "IconInfo",
        }
      }
      style={
        Array [
          Object {
            "height": 31.428571428571427,
            "width": 31.428571428571427,
          },
          Array [
            Object {
              "tintColor": "#3052FF",
            },
            null,
            null,
            null,
            null,
          ],
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "paddingVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Bold",
            "fontSize": 30,
            "textAlign": "center",
          },
          Object {
            "color": "#3052FF",
          },
          null,
          null,
          null,
          null,
          Object {
            "textAlign": "left",
          },
        ]
      }
    >
      label
    </Text>
  </View>
</View>
`;

exports[`[InfoMessage] should render with icon Warning 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 15,
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
      undefined,
      null,
      Object {
        "backgroundColor": "#FFF9ED",
        "borderColor": "#F8BF4C",
        "tintColor": "#F8BF4C",
      },
      null,
      null,
      null,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "paddingRight": 10,
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "IconInfo",
        }
      }
      style={
        Array [
          Object {
            "height": 31.428571428571427,
            "width": 31.428571428571427,
          },
          Array [
            null,
            Object {
              "tintColor": "#F8BF4C",
            },
            null,
            null,
            null,
          ],
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "paddingVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Bold",
            "fontSize": 30,
            "textAlign": "center",
          },
          null,
          Object {
            "color": "#F8BF4C",
          },
          null,
          null,
          null,
          Object {
            "textAlign": "left",
          },
        ]
      }
    >
      label
    </Text>
  </View>
</View>
`;
